crates/rpc-replay/tests/test_replay_block.rs

```diff
@@ -26,13 +27,28 @@ async fn test_replay_block() {
     let provider = JsonRpcClient::new(HttpTransport::new(
         Url::parse(provider_url.as_str()).expect("Could not parse provider url"),
     ));
-    let state_reader = AsyncRpcStateReader::new(provider, BlockId::Number(block_with_txs.block_number - 1));
+    let block_id = BlockId::Number(block_with_txs.block_number - 1);
+    let state_reader = AsyncRpcStateReader::new(provider, block_id);
     let mut state = CachedState::from(state_reader);
 
     let block_context = build_block_context(ChainId::Sepolia, &block_with_txs, StarknetVersion::V0_13_1);
 
-    for tx in block_with_txs.transactions.iter() {
-        let blockifier_tx = starknet_rs_to_blockifier(tx).unwrap();
+    // Workaround for JsonRpcClient not implementing Clone
+    let provider = JsonRpcClient::new(HttpTransport::new(
+        Url::parse(provider_url.as_str()).expect("Could not parse provider url"),
+    ));
+
+    let traces = provider.trace_block_transactions(block_id).await.expect("Failed to get block tx traces");
+    let gas_prices = GasPrices {
+        eth_l1_gas_price: 1u128.try_into().unwrap(), // TODO: update with 4844
```

Comment:
I don't understand the todo.

Reply:
I copied those values from a different file. I just removed the comment since it's a dummy value